datafusion/core/tests/order_spill_fuzz.rs

```diff
@@ -110,12 +107,23 @@ async fn run_sort(pool_size: usize, size_spill: Vec<(usize, bool)>) {
 /// with randomized i32 content
 fn make_staggered_batches(len: usize) -> Vec<RecordBatch> {
     let mut rng = rand::thread_rng();
-    let mut input: Vec<i32> = vec![0; len];
-    rng.fill(&mut input[..]);
-    let input = Int32Array::from_iter_values(input.into_iter());
-
-    // split into several record batches
-    let batch =
-        RecordBatch::try_from_iter(vec![("x", Arc::new(input) as ArrayRef)]).unwrap();
-    stagger_batch_with_seed(batch, 42)
+    let max_batch = 1024;
```

Comment:
Again, slicing batches causes the memory limiter to fail as it doesn't consider array slicing

Reply:
I believe stagger batches also adds zero length record batches, which was an important edge case to cover as I recall